### ISSUE TITLE
Duplicate column names in table panel - v7.0.0

### ISSUE THREAD

**Issue by Atomicbeast101** [grafana/grafana#24744]
**What happened**:
Combining two Prometheus metrics that has same labels gets added as separate columns:
![](https://i.imgur.com/AtSt4R8.png "Duplicate column names")

**What you expected to happen**:
Only unique column names in any table panels.

**How to reproduce it (as minimally and precisely as possible)**:
Add two metrics with same labels will show table with duplicate column names.

**Environment**:
- Grafana version: v7.0.0
- Data source type & version: Prometheus 2.17.1
- OS Grafana is installed on: Ubuntu v19.04
- User OS & Browser: Windows 10 & Chrome browser

Edited (5/26/2020) - Updating current version of Grafana

**Comment by baur** [comment]
You can hide them by transform 

![image](https://user-images.githubusercontent.com/2898361/82111501-ab532500-9767-11ea-80c3-5a05049a4548.png)

Sometimes this is useful, but require nested header cells: Related issue #24231

**Comment by baur** [comment]
related issue with table also #24231 (Ddd support for nested header cells, its very important in order to use transform -> JOIN)

**Comment by Atomicbeast101** [comment]
Hiding the field causes both duplicate columns to hide. Also trying to use JOIN feature to remove the duplicates and that didn't work as well, it basically moved one of the duplicated columns to first column section.

There doesn't seem to be an option to allow me to remove one of the duplicated columns.

**Comment by aocenas** [comment]
@Atomicbeast101 can you also post the prometheus query and any transforms you used for this table?

**Comment by Estrax** [comment]
Tried to reproduce this issue but wasn't able to do so, as it seems to be working properly. The only case when I see it working the way described in this issue would be when using outer join transformation.

**Comment by Atomicbeast101** [comment]
Sorry for the delay, here's the screenshot of the Prometheus query (there are different IPs under device label, in case you're wondering):
![](https://i.imgur.com/XRdIIRS.png "")

Grafana query section:
![](https://i.imgur.com/tNHc1E2.png "")

Grafana transform section:
![](https://i.imgur.com/Q49tg8z.png "")

Trying to use outer join on interface doesn't work as it merges both metrics into one instead of keeping them separate.

**Comment by Atomicbeast101** [comment]
I forgot to mention, I'm doing this on v7.0.0 stable release.

**Comment by aocenas** [comment]
So to me it seems the best solution would be to be able to hide one of those columns, so the transform should not work by the column name only but based on some id like `dataframe + column name` to make it at least internally non ambiguous.

@torkelo or @dprokop this seems like an issue with the transforms but maybe such issue already exists somewhere?

Also this affects both Organize fields and Filter by name transforms although the later one at least shows that there are 2 fields with the same name.
![Screenshot from 2020-05-27 17-23-42](https://user-images.githubusercontent.com/1014802/83039906-fbbb6400-a03e-11ea-9219-5a31a35790c6.png)


I assume we could try to deduplicate on Prometheus data source level but I am not sure how much assumptions we can make around column names and if the same column name from one datasource will for sure always mean also duplication of the values in that column and also I suspect this may not be prometheus only issue, so fixing it on transform level would probably be more universal.

**Comment by torkelo** [comment]
Think the problem is that we have not added support for "merge tables" that the old table panel supported, but reading the code we where unable to understand how it works or how to replicate it for data frames

**Comment by Atomicbeast101** [comment]
@torkelo That may have been the feature I was using in the old version of table to merge these two metrics together.

Believe I had to include "+0" by both metrics in the query section in order to achieve this in older version.

**Comment by aocenas** [comment]
@torkelo can't we just process the duplicate names when doing joins to something like `app (1)`, `app (2)` or something like that to make them target-able by the other transforms?

**Comment by dwilson6** [comment]
What about having an option in the outer join transform for whether to merge duplicate column names or whether to append the `refId` to them?

**Comment by git4fred** [comment]
Hi, I've got the same issue. Until this issue is fix under grafana 7, I use `label_rename` in my query.

In my case, I want to join metrics from `kube_poddisruptionbudget_status_current_healthy` and
`kube_poddisruptionbudget_status_expected_pods`, and join them on `poddisruptionbudget` label.

As a result, I have duplicate column on `namespace` label. To solve this, I use these two queries:

```
# first query
sum without﻿(﻿)﻿(﻿label_replace﻿(﻿kube_poddisruptionbudget_status_pod_disruptions_allowed﻿﻿,﻿"ns"﻿,﻿"$﻿1﻿"﻿,﻿"namespace"﻿,﻿"﻿(﻿.﻿*﻿)﻿"﻿)﻿)

# second query
kube_poddisruptionbudget_status_current_healthy
```

Then I transform:
- with an outer join on `poddisruptionbudget` label
- with an organize fields on which I hide the `namespace` label, and rename `ns` label to `namespace`

Hope this help,

**Comment by KalyanMahesh** [comment]
Hi,

I got same issue. I'm using v7.0.5 and with data source MS SQL database

My queries are

Query 1 : SELECT Name , Nodes, VMs, vCPU from Tab1
Query 2 : SELECT 'Total' as Name,SUM(Nodes) as Nodes,SUM(VMs) as VMs,SUM(vCPU) as vCPU from Tab1

with Outer Join by Name in Transform

Output table is showing 2 columns for each identifier.

**Comment by Atomicbeast101** [comment]
@KalyanMahesh I believe this is being fixed in v7.1.0 update?

**Comment by macgaver** [comment]
I am on 7.1.3 and the issue is still there

**Comment by dykow** [comment]
Having the same issue on 7.1.0
![image](https://user-images.githubusercontent.com/16824096/90009102-bf8ebb80-dc9d-11ea-8483-f3ef1d3175a9.png)
![image](https://user-images.githubusercontent.com/16824096/90009158-d7fed600-dc9d-11ea-9d7f-5ecef4959eb9.png)

**Comment by dykow** [comment]
Having the same issue on 7.1.0
It doesn't happen for label I do outer join on, which is `instance`.

![image](https://user-images.githubusercontent.com/16824096/90009465-670bee00-dc9e-11ea-9dea-41a4f272e3db.png)

![image](https://user-images.githubusercontent.com/16824096/90009232-ffee3980-dc9d-11ea-8531-892ee55ab42f.png)

**Comment by Atomicbeast101** [comment]
@dykow @macgaver Did you use merge option under Transform?

**Comment by dykow** [comment]
@Atomicbeast101 I tried, but it produces empty fields in tabel.
Those 3 rows should be joined into one:
![image](https://user-images.githubusercontent.com/16824096/90369090-6b078980-e06b-11ea-85f0-c760754020f1.png)

**Comment by beingyash** [comment]
I am facing the same issue with elasticsearch metric name when two queries have the same metric like 'Count' or 'Unique Count'. Both queries have been returning different data in each column and i'm unable to rename those columns individually. 'Organize fields' option renames both the columns on attempting.

**Comment by Deepak1100** [comment]
we are facing the same issue
we are using table for prometheus histogram metrics

**Comment by HenleyKuang** [comment]
> Hi, I've got the same issue.
> 
> Until this issue is fix under grafana 7, I use `label_replace` in my query (note: `label_replace` add a label, not rename the existing one).
> 
> In my case, I want to join metrics from `kube_poddisruptionbudget_status_current_healthy` and
> `kube_poddisruptionbudget_status_expected_pods`, and join them on `poddisruptionbudget` label.
> 
> As a result, I have duplicate column on `namespace` label. To solve this, I use these two queries:
> 
> ```
> # first query
> label_replace﻿(﻿kube_poddisruptionbudget_status_pod_disruptions_allowed﻿﻿,﻿"ns"﻿,﻿"$﻿1﻿"﻿,﻿"namespace"﻿,﻿"﻿(﻿.﻿*﻿)﻿"﻿)﻿
> 
> # second query
> kube_poddisruptionbudget_status_current_healthy
> ```
> 
> Then I transform:
> 
> * with an outer join on `poddisruptionbudget` label
> * with an organize fields on which I hide the `namespace` label, and rename `ns` label to `namespace`
> 
> Hope this help,

Although it's not elegant, this was the solution that worked for me

**Comment by Telokis** [comment]
Maybe I'm not using the `label_replace` workaround properly but it doesn't return anything for me.

`label_replace(node_uname_info{customer=~"$customer", stage=~"$stage"}﻿﻿,﻿"cust"﻿,﻿"$﻿1﻿"﻿,﻿"customer"﻿,﻿"﻿(﻿.﻿*﻿)﻿")`

**Comment by torkelo** [comment]
I think this should be fixed in 7.2

**Comment by stwellman** [comment]
This was fixed in 7.2 but is back in 7.4.
